src/rosparam_shortcuts.cpp

```diff
@@ -255,20 +294,63 @@ std::string getDebugArrayString(std::vector<std::string> values)
 
 bool convertDoublesToEigen(const std::string &parent_name, std::vector<double> values, Eigen::Affine3d &transform)
 {
-  if (values.size() != 6)
+  if (values.size() == 6)
+  {
+
+    // This version is correct RPY
+    Eigen::AngleAxisd roll_angle(values[3], Eigen::Vector3d::UnitX());
+    Eigen::AngleAxisd pitch_angle(values[4], Eigen::Vector3d::UnitY());
+    Eigen::AngleAxisd yaw_angle(values[5], Eigen::Vector3d::UnitZ());
+    Eigen::Quaternion<double> quaternion = roll_angle * pitch_angle * yaw_angle;
+
+    transform = Eigen::Translation3d(values[0], values[1], values[2]) * quaternion;
+
+    return true;
+  }
+  else if (values.size() == 7)
+  {
+    double x = values[0];
+    double y = values[1];
+    double z = values[2];
+    double qw = values[3];
+    double qx = values[4];
+    double qy = values[5];
+    double qz = values[6];
```

Comment:
don't instantiate all this memory - pass the ``values`` directly into the Eigen() calls